Source/Bogus/Extensions/Denmark/ExtensionsForDenmark.cs

```diff
@@ -8,19 +11,169 @@ public static class ExtensionsForDenmark
       /// <summary>
       /// Danish Personal Identification number
       /// </summary>
-      public static string Cpr(this Person p)
+      /// <param name="p">The holder.</param>
+      /// <param name="validChecksum">
+      ///   Indicates whether the generated CPR number should have a valid checksum or not.
+      /// </param>
+      public static string Cpr(this Person p, bool validChecksum = true)
       {
          const string Key = nameof(ExtensionsForDenmark) + "CPR";
-         if( p.context.ContainsKey(Key) )
+         if (p.context.ContainsKey(Key))
          {
             return p.context[Key] as string;
          }
 
+         /*
+             DDMMYY-XXXX
+             | | |  |  
+             | | |  |
+             | | |  |
+             | | |  |----> (X)Individual number
+             | | |-------> (Y)Year (last two digits)
+             | |---------> (M)Month
+             |-----------> (D)Day
+
+             The individual number has to be even for women and odd for men.
+
+             As of 2007 there is no longer a requirement for a checksum with a modulo algorithm.
+         
+            https://cpr.dk/cpr-systemet/opbygning-af-cpr-nummeret
+               
+            https://da.wikipedia.org/wiki/CPR-nummer
+                  
+            https://www.cprgenerator.net/metode
+         */
+
          var r = p.Random;
-         var final = $"{p.DateOfBirth:ddMMyy}-{r.Replace("####")}";
+         string birthDate = $"{p.DateOfBirth:ddMMyy}";
+         string individualNumber;
+         string checksum;
+         bool hasValidChecksum;
+
+         if (validChecksum)
+         {
+            do
+            {
+               individualNumber = GenerateIndividualThreeDigitNumber(r, p.DateOfBirth.Year);
+               hasValidChecksum = GenerateChecksum(birthDate, p.Gender, individualNumber, out checksum);
+            } while (!hasValidChecksum);
+         }
+         else
+         {
+            checksum = string.Empty;
+            individualNumber = GenerateIndividualFourDigitNumber(r, p.Gender, p.DateOfBirth.Year);
+         }
+
+         var final = $"{birthDate}-{individualNumber}{checksum}";
```

Comment:
It would be nice to have a parameter to toggle this dash off.

Reply:
The `includeDash` parameter has been added in the latest commit